pkg/catalog/inbound_traffic_policies.go

```diff
@@ -61,15 +63,17 @@ func (mc *MeshCatalog) listInboundPoliciesFromTrafficTargets(upstreamIdentity id
 // listInboundPoliciesForTrafficSplits loops through all SMI TrafficTarget resources and returns inbound policies for apex services based on the following conditions:
 // 1. the given upstream identity matches the destination specified in a TrafficTarget resource
 // 2. the given list of upstream services are backends specified in a TrafficSplit resource
-func (mc *MeshCatalog) listInboundPoliciesForTrafficSplits(upstreamIdentity identity.K8sServiceAccount, upstreamServices []service.MeshService) []*trafficpolicy.InboundTrafficPolicy {
+func (mc *MeshCatalog) listInboundPoliciesForTrafficSplits(upstreamIdentity identity.ServiceIdentity, upstreamServices []service.MeshService) []*trafficpolicy.InboundTrafficPolicy {
```

Comment:
Could you add in comments what `upstreamIdentity` should look like: `foo.bar` vs `foo.bar.cluster.local`.

This comment applies to any function that takes a `ServiceIdentity` type as an argument.

Reply:
@shashankram I do not agree that there should be such comment. We created `ServiceIdentity` as a type to give us the flexibility to define these things in the type itself.

I comment with the sample string would only be needed because `ServiceIdentity` is still just a string alias.  When we design and implement something more complex `type ServiceIdentity struct` we would have this information encoded in the struct.

So in the short term I could go around sprinkling comments, but these would be relevant for a short period of time while `ServiceIdentity` is a string.

Reply:
I think there needs to be a comment, otherwise authors not familiar with the concept of ServiceIdentity are going to wonder what should be passed as an argument to functions that accept a `ServiceIdentity` type (`foo.bar` vs `foo.bar.cluster.local`). Even just adding this to the interface will suffice for now, so its clear what the catalog interface expects at this moment.

Reply:
@shashankram I'll add the comment and look forward to sophisticating `ServiceIdentity` from a `string` to a `struct`, which would be explicit about the components of this string (now implied and relying on comments). We can discuss and tackle this going forward with the following GitHub Issue:
  - [ ] **identity: Change (sophisticate) ServiceIdentity from a string to a struct{} so it can capture more context** #3188

Reply:
Thanks @draychev, the new issues make sense.